### ISSUE TITLE
F-Droid package in a recent version

### ISSUE THREAD

**Issue by jannis701** [kiwix/kiwix-android#3651]
The kiwix-package on F-Droid is Version 2.4 from 2018. It would be nice to have a recent version there (not necessarily a nightly build as suggested in https://github.com/kiwix/kiwix-android/issues/1777 ) 

**Is your feature request related to a problem? Please describe.**
The Problem is, that it is not possible to get convenient updates for the app via f-droid and therefore not at all without having a google-account.

**Describe the solution you'd like**
a recent kiwix-Version sould be available in the standard f-droid appstore (without adding any custom repo)

**Additional context**
Also the "Get it on F-Droid"-Button at https://kiwix.org/en/applications/ points to the F-Droid-APK instead of https://f-droid.org/de/packages/org.kiwix.kiwixmobile/

**Comment by Jaifroid** [comment]
There is this IzzyOnDroid F-Droid package which is 3.8.0: https://apt.izzysoft.de/fdroid/index/apk/org.kiwix.kiwixmobile.

But definitely, the main F-Droid site should be updated too.

**Comment by Popolechien** [comment]
I've updated the website link to izzydroid (it still bears the F-droid logo, not sure this is kosher but so be it).

**Comment by Jaifroid** [comment]
Posters (I think on Reddit, from memory) have recommended that fork of F-Droid. Personally, I'd suggest this should be a stop-gap measure, and it would be much better to update the official F-Droid asap if it is no longer blocked by F-Droid policies!

**Comment by KaKi87** [comment]
Hello,
Any news on this ?
Thanks

**Comment by kelson42** [comment]
Like written, not yet, but is for current milestone

**Comment by JensKorte** [comment]
A workaroud would be if https://github.com/kiwix/kiwix-android/releases would still provide the apk since then https://apt.izzysoft.de/fdroid/index.php?repo=iod could collect and link to them.

**Comment by IzzySoft** [comment]
> it still bears the F-droid logo, not sure this is kosher but so be it

You can [pick a proper badge here](https://gitlab.com/IzzyOnDroid/repo/#if-my-app-is-available-here-do-you-have-a-badge-i-can-use-to-link-to-it)

> A workaroud would be if [kiwix/kiwix-android/releases](https://github.com/kiwix/kiwix-android/releases) would still provide the apk

That would be very good, as since 11/2023 (said release 3.8.0) they are missing. For starters, maybe manually add the armeabi to the latest release? The IzzyOnDroid updater would then fetch that automatically (I just noticed it had been missing since, which is why I chime in here).

**Comment by JensKorte** [comment]
As a workaround I use [Obtainium](https://f-droid.org/de/packages/dev.imranr.obtainium.fdroid/) and add https://download.kiwix.org/release/kiwix-android/ as a source. Installing worked I am waiting for the first update.

**Comment by Jaifroid** [comment]
@mossroy Hello, long time no see indeed! I thought I'd jump in because I recently asked about this, so know a little about it (no, I'm still focused on Kiwix JS, not Android, but I got notified here because I commented further up).

You raise two issues really: 1. The F-Droid entry appears to be interfering with the Play Store version on AppLounge; and 2. The APK conflicts with a version installed from the Store(s).

The first issue should have been resolved (not in a satisfactory way) by https://github.com/kiwix/kiwix-android/issues/3672. Maybe that (temporary) suspension hasn't filtered through to AppLounge yet?

The second issue should have been resolved by https://github.com/kiwix/kiwix-android/issues/3933 (giving a different AppID to the APK compared to the Play Store). But that was only closed last week I think. So it won't have filtered through yet either.

**Comment by Jaifroid** [comment]
@IzzySoft Some time ago, I opened https://github.com/kiwix/kiwix-android/issues/3830, but it got closed because it's not clear who should do the update to the [Kiwix entry](https://apt.izzysoft.de/fdroid/index/apk/org.kiwix.kiwixmobile) on the IzzyOnDroid store. @IzzySoft, who would normally update that entry on your Store? Do you have automatic rules that could update it from the [APK version](https://download.kiwix.org/release/kiwix-android/kiwix.apk)? That's a permalink to the latest version -- does that cause issues?

**Comment by IzzySoft** [comment]
> Who would normally update that entry on your Store?

Depends on which part. Description and graphics? If you'd establish Fastlane structures here in your repo (see the [IzzyOnDroid Fastlane Documentation](https://gitlab.com/IzzyOnDroid/repo/-/wikis/Fastlane) for reference on that), the updater would fetch that along with each release as soon as it's enabled. Background for "enabling" is I'd check for the best options matching what you've set up, for the best possible presentation (e.g. your `full_description.txt` could be proper HTML, or render as Markdown – or as last resort we'd have to go with "plain text", just replacing line breaks with `<br>`).

> Do you have automatic rules that could update it from the [APK version](https://download.kiwix.org/release/kiwix-android/kiwix.apk)? That's a permalink to the latest version -- does that cause issues?

Let me check… Using that variant would make reproducible builds impossible (see [Reproducible Builds, special client support and more in our repo](https://android.izzysoft.de/articles/named/iod-rbs-mirrors-clients) which was published today, and our [hints on reproducible builds](https://gitlab.com/IzzyOnDroid/repo/-/wikis/Reproducible-Builds)). Your app is pure Kotlin, so if the app is attached to the releases here it could be made RB. Apart from that, the server sends a Last_Modified header, so that could be used. But full stop there: the APK is 90+ MB, that's 3 times the per-app limit at IoD. Native libs are just stored (not compressed) I see, so a single ABI (armeabi or arm64) might be close enough to that limit though.

**Comment by mnalis** [comment]
So, what is the status with f-droid?  Looking at https://gitlab.com/fdroid/fdroiddata/-/issues/3024 from a year ago, it was waiting for migration from objectbox to Room in https://github.com/orgs/kiwix/projects/22/views/1, which looks mostly complete (just needs some cleanup of old unused classes in https://github.com/kiwix/kiwix-android/issues/3106?)

Is anything else needed for Kiwix to go into F-droid again?

**Comment by MohitMaliFtechiz** [comment]
> Is anything else needed for Kiwix to go into F-droid again?

@mnalis Please see https://github.com/kiwix/kiwix-android/issues/3106#issuecomment-2834190092.

**Comment by kelson42** [comment]
@MohitMaliFtechiz FDroid version should not include ObjectBox code, any configuration migration code, and ot seems we are ready to move forward on this. Can you confirm?

**Comment by MohitMaliFtechiz** [comment]
@licaon-kter We have removed all the code from our project related to the objectbox(except migration code and dependency), which is required for migration. Otherwise, the project would not compile if we removed the dependency.

I saw that Objectbox provided the generator source code on GitHub https://github.com/objectbox/objectbox-java/issues/560#issuecomment-1980184740, and the linked issue in https://github.com/kiwix/kiwix-android/issues/309 is closed.
Additionally, you know why this dependency is required to avoid runtime issues as mentioned in the conversation in https://gitlab.com/fdroid/fdroidserver/-/merge_requests/1150. Also, you have added this in your scanner.py https://gitlab.com/fdroid/fdroidserver/-/merge_requests/1150/diffs#313e88ef2ffda23f21fe2a93447bbc5a553a59c3_95_97.  So, does that mean we can relaunch our application on fdroid?

**Comment by licaon-kter** [comment]
rules now live here: https://gitlab.com/fdroid/fdroid-suss/-/blob/master/suss/io.objectbox.yml

@MohitMaliFtechiz which release shall we test?

**Comment by MohitMaliFtechiz** [comment]
@licaon-kter We should test with the version [3.14.0](https://github.com/kiwix/kiwix-android/releases/tag/3.14.0) it is latest stable release with Android 15 support. According to your rules, we are using the Gradle plugin in this version, which is itself FOSS, so objectbox code removal will not have an effect on release(which we have done in `4.0.0`). So we should test with the stable release version(3.14.0).

**Comment by licaon-kter** [comment]
3.14.0 builds 

```
...
2025-08-21 11:53:09,180 DEBUG: Checking build/org.kiwix.kiwixmobile/app/build/outputs/apk/release/kiwix-universal-release.apk
2025-08-21 11:53:09,196 INFO: Scanning APK with dexdump for known non-free classes.
2025-08-21 11:53:09,407 DEBUG: > /opt/android-sdk/build-tools/34.0.0/dexdump /tmp/tmpq_ii0u24/classes.dex
2025-08-21 11:53:18,484 DEBUG: Problem: found class 'io/objectbox/flatbuffers/Utf8Safe'
2025-08-21 11:53:18,486 DEBUG: Problem: found class 'io/objectbox/converter/LongFlexMapConverter'
2025-08-21 11:53:18,521 DEBUG: Problem: found class 'io/objectbox/exception/DbExceptionListener'
...
2025-08-21 11:53:19,752 INFO: Scanning APK for extra signing blocks.
2025-08-21 11:53:21,606 WARNING: Requested API level 35 is larger than maximum we have, returning API level 28 instead.
2025-08-21 11:53:21,847 CRITICAL: Found 51 problems in build/org.kiwix.kiwixmobile/app/build/outputs/apk/release/kiwix-universal-release.apk
```

@MohitMaliFtechiz what's the plan, timeline, etc?

we used https://download.kiwix.org/release/kiwix-android/VERSION_INFO to gather version info in the past, but that's abandoned?

Used recipe:
```
  - versionName: 3.14.0
    versionCode: 7231501
    commit: f6aef1b7ea809753d90e05ca8f000009f6151369
    gradle:
      - yes
    output: app/build/outputs/apk/release/kiwix-universal-release.apk
    scandelete:
      - contrib/move-string-resource
    ndk: 27.3.13750724
    gradleprops:
      - disableSigning
```

not using `subdir: app` because we'd need to patch https://github.com/kiwix/kiwix-android/blob/3.14.0/app/build.gradle.kts#L144-L149 instead

**Comment by MohitMaliFtechiz** [comment]
> @MohitMaliFtechiz what's the plan, timeline, etc?

@licaon-kter We have removed all the objectbox code from our project except the entity and dependency, which is required for migration in https://github.com/kiwix/kiwix-android/pull/4396(in version 4.0.0). So for now, we can not remove it. But we could remove this migration code in the future when all the users have migrated to the room database.

> we used https://download.kiwix.org/release/kiwix-android/VERSION_INFO to gather version info in the past, but that's abandoned?

Yes, I have remembered that we had left the update of this file because we could not update the application on fdroid, and updating this file could trigger the workflow of fdroid. But we will re-enable it once our application is published on fdroid for future updates.

> - versionName: 3.14.0
    versionCode: 7231501
    commit: f6aef1b7ea809753d90e05ca8f000009f6151369
    gradle:
      - yes
    output: app/build/outputs/apk/release/kiwix-universal-release.apk
    scandelete:
      - contrib/move-string-resource
    ndk: 27.3.13750724
    gradleprops:
      - disableSigning

@licaon-kter The version code you are building is wrong. The version `3.14.0` has the versionCode `7231312`.  Are you building on the application on `3.14.0` release commit? We have added the functionality to build the application for a specific date in https://github.com/kiwix/kiwix-android/issues/4120 for a reproducible build. Please consider this when you build the application in the future.

<img width="1774" height="664" alt="Image" src="https://github.com/user-attachments/assets/19480793-22fe-43dc-a22a-267372e29acb" />

**Comment by kelson42** [comment]
@MohitMaliDeveloper Only ps version should have this migration code. All other versions, and in particular the FDroid version, must be 100,% free of it.

**Comment by MohitMaliFtechiz** [comment]
@kelson42 Okay, but for a noticeable impact of this, i want to highlight some points:

We have introduced the `.stanalone` APK in `3.12.0`(which is currently available everywhere, except Google playStore), and before that, we have migrated the objectbox to room for (Notes, history, recent searches in version 3.11.0), and bookmark in the release 3.10.0(for that user has the option to export and import). But for the local library, we recently migrated it to libkiwix, and due to the absence of migration. There will not be available the previous list of ZIM files users need to refresh the library to see them.

There will be only this effect. Apart from this, due to a different app ID, there will not be much impact on the users. If they want to update the application, they need to download a new application due to a different app ID. Also, which users have already installed the application with the new app ID (there is already a room database instead of ObjectBox).

@kelson42 For releasing the application on fdroid, we should go for a `.standalone` App like we published the application on [IZZYOnDroid](https://apt.izzysoft.de/fdroid/index/apk/org.kiwix.kiwixmobile.standalone) or we should publish with the old version.

**Comment by mnalis** [comment]
>>     we used https://download.kiwix.org/release/kiwix-android/VERSION_INFO to gather version info in the past, but that's abandoned?


> Yes, I have remembered that we had left the update of this file because we could not update the application on fdroid, and updating this file could trigger the workflow of fdroid. But we will re-enable it once our application is published on fdroid for future updates.

I think you can enable updating of `VERSION_INFO` already, as AFAIK there is no risk at the moment that any workflow would be triggered on F-droid, as it has [disabled any updates](https://gitlab.com/fdroid/fdroiddata/-/blob/master/metadata/org.kiwix.kiwixmobile.yml#L145-146) for `org.kiwix.kiwixmobile`.

 > and in particular the FDroid version, must be 100,% free of it.

Absolutely, FDroid version should not have _any_ `ObjectBox` code linked in. Is there a build recipe/flavour which can build without that  `ObjectBox` code?

**Comment by MohitMaliFtechiz** [comment]
> I think you can enable updating of VERSION_INFO already, as AFAIK there is no risk at the moment that any workflow would be triggered on F-droid, as it has [disabled any updates](https://gitlab.com/fdroid/fdroiddata/-/blob/master/metadata/org.kiwix.kiwixmobile.yml#L145-146) for org.kiwix.kiwixmobile.

@mnalis Okay, we will reenable it.

> Absolutely, FDroid version should not have any ObjectBox code linked in. Is there a build recipe/flavour which can build without that ObjectBox code?

Not yet, currently all the flavours have objectbox dependency, but I am making an `fdroid` which will be free of objectbox.

**Comment by mnalis** [comment]
So, with #4397 merged, if I understand correctly, F-droid should be able to build version of Kiwix without any non-free code? 

What is the next step needed towards having new version of Kiwix on F-droid?

**Comment by MohitMaliFtechiz** [comment]
@mnalis We are publishing the nightlies on fdroid, see https://github.com/kiwix/kiwix-android/issues/1777. And after that, we will release the full version 4.0.0 on fdroid. Currently, in `4.0.0`, there are some tickets left to do. After that, we will make a release and will upload the app to the official fdroid.

**Comment by mnalis** [comment]
> We are publishing the nightlies on fdroid, see https://github.com/kiwix/kiwix-android/issues/1777.

Thanks for the pointer @MohitMaliFtechiz! Unless I'm misinterpreting, that one seems to be about publishing your own custom fdroid-compatible repository for nightlies? To clarify, my interest is primarily about inclusion in official repository on f-droid.org (I like them doing extra checks after the original developers; as it has saved me several times already from FOSS apps unintentionally including non-free binary blobs). 

IOW, when I (or anyone else) as a user install unmodified F-droid client (without doing any extra configuration modification), and search for "kiwix", I'd like to see result for recent kiwix version.

> After that, we will make a release and will upload the app to the official fdroid.

Yes, the official repository at f-droid.org is what I was interested about. So, if I understand correctly, I should wait for `4.0.0` to be released on GitHub, and then recheck on f-droid.org inclusion status, yes?

**Comment by IzzySoft** [comment]
> (I like them doing extra checks after the original developers; as it has saved me several times already from FOSS apps unintentionally including non-free binary blobs

Yupp, that's why we have those extra checks as well – and why Kiwix is currently available [at IzzyOnDroid](https://apt.izzysoft.de/fdroid/packages/org.kiwix.kiwixmobile.standalone) 😉 For the checks performed here, see e.g. [What about security?](https://apt.izzysoft.de/fdroid/index/info#security) in our info page.

So until Kiwix becomes available at F-Droid, you can get the official builds of the developers (cross-checked by our framework) by adding the IzzyOnDroid repo to your [F-Droid client](https://android.izzysoft.de/applists/category/named/apps_markets#group_1181) (if you wonder about the alternative clients: NeoStore, and soon also Droid-ify, already include a feature folks are asking about for years: download statistics, and ordering apps by "popularity" (how often they are downloaded). They also show the reproducibility status of apps (though just for those built on one of our builders).

**Comment by mnalis** [comment]
>>    (I like them doing extra checks after the original developers; as it has saved me several times already from FOSS apps unintentionally including **non-free binary blobs**

I was under impression that IzzyOnDroid was allowing the apps to have some "smaller" non-free binary blobs included in the apps? Because, that is exactly what I want to **avoid**, and main F-droid repository ensures that for me (even when it originally slips beyond the the app devs radar like with Kiwix[^1])

> Yupp, that's why we have those extra checks as well – and why Kiwix is currently available [at IzzyOnDroid](https://apt.izzysoft.de/fdroid/packages/org.kiwix.kiwixmobile.standalone) 😉

That link returns 404... There is [this one](https://apt.izzysoft.de/fdroid/index/apk/org.kiwix.kiwixmobile.standalone), but when I look at it, it seems that ObjectBox library is linked in? I'm interested in 100%-FOSS Kiwix (i.e. one that is eligible for main F-droid repo, and which I could also compile from scratch without _**any**_ non-free software **_anywhere_** in the toolchain). 

Thus my inquiring when such fully-FOSS Kiwix can be expected at F-droid.org main repository (as that is only place that I am aware of that attempts to verify / guarantee such 100% "FOSSness")[^2]

[^1]: or the devs don't seem to care [enough], like with [EveryDoor recently](https://github.com/Zverik/every_door/issues/975) 😢 

[^2]:  Or is there a way I can configure IzzyOnDroid repo to **only**  ever offer such apps which are 100% FOSS (i.e. no GMS, no ObjectBox, no firebase etc.) and get such version of Kiwix there?

**Comment by IzzySoft** [comment]
> I was under impression that IzzyOnDroid was allowing the apps to have some "smaller" non-free binary blobs

See the [IzzyOnDroid App Inclusion Policy](https://gitlab.com/IzzyOnDroid/repo/-/wikis/Inclusion-Policy) for details. And if there are such exceptions (which get fewer and smaller all the time), those are clearly marked with the `NonFreeComp` ("contains non-free components") anti-feature.

> Or is there a way I can configure IzzyOnDroid repo to only ever offer such apps which are 100% FOSS

If you use the 100% official F-Droid client, simply tell it you don't want any apps marked with the `NonFreeComp` anti-feature 😉

**Comment by mnalis** [comment]
> If you use the 100% official F-Droid client, simply tell it you don't want any apps marked with the `NonFreeComp` anti-feature 😉

Thanks @IzzySoft , but I am confused. If I select that I don't want any `NonFreeComp` apps[^1], would IzzyOnDroid show me Kiwix, or not? And if it would show it me, would it really be "clean" version (without any ObjectBox binary blobs), or not[^2]? 

---

Because, I'm looking where I can get _**fully FLOSS**_ Kiwix `.apk` today (i.e. one without _any_ proprietary ObjectBox baggage linked in). **Does it exist somewhere**? (on IzzyOnDroid as the post above seems to imply? Or on some separate Kiwix F-droid nightly repo? GitHub releases?) 

(Or it does not yet exist anywhere, and I should wait for `4.0.0` release?)


[^1]:  which is something I'm _additionally_ also having problems with -- how to select that in official F-droid client that I don't want `NonFreeComp`; see [IzzyOnDroid #883](https://gitlab.com/IzzyOnDroid/repo/-/issues/883) about that
[^2]: see [IzzyOnDroid #884](https://gitlab.com/IzzyOnDroid/repo/-/issues/884) about that, to not spam kiwix repo

**Comment by mnalis** [comment]
> Not yet, currently all the flavours have objectbox dependency, but I am making an `fdroid` which will be free of objectbox.

Is there an issue which tracks that progress @MohitMaliFtechiz ?

**Comment by MohitMaliFtechiz** [comment]
@mnalis For f-droid nightly here we have an issue https://github.com/kiwix/kiwix-android/issues/1777, and see https://github.com/kiwix/kiwix-android/issues/1777#issuecomment-3244870413. We wil release the app on F-droid once we release the `4.0.0`, but before releasing we will publish the nightlies on fdroid.

**Comment by IzzySoft** [comment]
> without any ObjectBox binary blobs

Please correct me if I'm wrong, but the issue with the BLOBs is solved, isn't it? According to [this issue](https://github.com/objectbox/objectbox-java/issues/560), it is.

> how to select that in official F-droid client that I don't want `NonFreeComp`

As pointed out: an issue for [fdroidclient](https://gitlab.com/fdroid/fdroidclient/-/issues), which seems to ignore "custom anti-features" set by a repository (and defined in its index), and only acknowledges the set uses by F-Droid.org. `NonFreeComp` and `SourceGone` are 2 anti-features used at IzzyOnDroid, but not at F-Droid – and both are missing in the client's "check list".

**Comment by licaon-kter** [comment]
Are they in "Other" basically? @IzzySoft

**Comment by IzzySoft** [comment]
I don't know – that could of course be (what other sense should "other" have "otherwise"). @mnalis can you verify with some app that has NonFreeComp or SourceArchived? And would that then answer this part of your question?

**Comment by mnalis** [comment]
> > without any ObjectBox binary blobs
> 
> Please correct me if I'm wrong, but the issue with the BLOBs is solved, isn't it? According to [this issue](https://github.com/objectbox/objectbox-java/issues/560), it is.

I think you're wrong. That issue was only about opensourcing Objectbox gradle plugin part IIRC[^1], and it links to other closed source ObjectBox thingies (but you have to read the whole ticket for all details). The other parts of ObjectBox are still closed source AFAICT, see e.g. https://github.com/objectbox/objectbox-java/issues/1102 as well as their [What about Open Source?](https://objectbox.io/faq/#license-pricing) section on their  website.

And since current version of https://apt.izzysoft.de/fdroid/index/apk/org.kiwix.kiwixmobile.standalone under "libraries" still lists ObjectBox, I would assume it (any any other package using ObjectBox) should be tagged `NonFreeComp` as it includes non-free binary blobs @IzzySoft  

> I don't know – that could of course be (what other sense should "other" have "otherwise"). [@mnalis](https://github.com/mnalis) can you verify with some app that has NonFreeComp or SourceArchived? And would that then answer this part of your question?

Yeah, "other" will hide all apps having any antifeature not listed above. But that doesn't work for me, because:
- as noted above, Kiwix is not properly marked with `NonFreeComp` in IzzyOnDroid (see [IzzyOnDroid #884](https://gitlab.com/IzzyOnDroid/repo/-/issues/884))
- such an aggregation of wildly different AntiFeatures is too wide a sledgehammer for me. [^2]. If someone could recommend f-droid client which allow each antifeature to be selected individually (including all 3rd-party AFs like `NonFreeComp`), I'd be interested in trying that out, 
- regardless of both issues above, I don't want to install Kiwix until its `.apk` is fully FOSS, so solving above issues would only help me insofar as to not seeing Kiwix in the list (so I won't install nonfree version of it by accident); while I what would instead like to be able to do is run fully FOSS version of Kiwix.

**TL;DR:** AFAIK there is still no fully-FOSS `.apk` release of Kiwix publicly available, and since that is the only thing that I am interested in running on my mobile devices, I am still sad. 😿 

[^1]:  which they were legally obliged to provide (due to Gradle GPLv3 requirements), so they couldn't leave it as closed source (which they would've preferred).
[^2]:  For example, I might be fine with running fully FOSS older version of some software, even if it is no longer being developed (at all, or as FOSS, e.g. _"developers sold out"_) - I just won't be recommending it to other people, as they could fall into trap of upgrading to non-free version available elsewhere.
